internal/certificateprovider/certificateproviderpage/enter_reference_number.go

```diff
@@ -40,13 +49,35 @@ func EnterReferenceNumber(tmpl template.Template, shareCodeStore ShareCodeStore,
 						data.Errors.Add("reference-number", validation.CustomError{Label: "incorrectReferenceNumber"})
 						return tmpl(w, data)
 					} else {
-						return err
+						return fmt.Errorf("error getting shareCode: %w", err)
+					}
+				}
+
+				lpa, err := lpaStoreClient.Lpa(r.Context(), shareCode.LpaUID)
+				if err != nil && !errors.Is(err, lpastore.ErrNotFound) {
+					return fmt.Errorf("error getting LPA from LPA store: %w", err)
+				}
+
+				if lpa != nil && lpa.CertificateProvider.Channel.IsPaper() && !lpa.CertificateProvider.SignedAt.IsZero() {
+					redirectTo := page.PathCertificateProviderYouHaveAlreadyProvidedACertificateLoggedIn
+
+					if results.Empty() {
+						if err = sessionStore.ClearLogin(r, w); err != nil {
```

Comment:
I missed this the first time I looked, why are we logging them out here?

Reply:
I interpreted the [design](https://www.figma.com/design/OML6o3PgghlnhYBYhX8g4Z/Double-acting-attempt?node-id=16-278) to mean the journey should end if they don't have any other LPAs in their account. I started with something manual to hide the nav, but if we wanted them to close the browser, it felt easiest to log them out.